File: api/models/product.py
from pydantic import BaseModel
from typing import Optional
from datetime import date

class Product(BaseModel):
    id: int
    name: str
    price: float
    family_id: int
    format: str
    dots: int
    last_changed: str
    active: int

    class Config:
        orm_mode = True

class ProductUpdate(BaseModel):
    #id: int
    name: str
    price: float
    #family_id: int
    format: str
    dots: int
    #last_changed: str // no es necesario ya que se actualiza automaticamente
    #active: int

    class Config:
        orm_mode = True

File: apps/server/models/Product.js
const mongoose = require('mongoose');

//id, name, price, family_id, format, dots, last_changed, active
const productSchema = new mongoose.Schema({
    plu: { type: Number, required: true },
    name: { type: String, required: true },
    price: { type: Number, required: true },
    subfamily_id: { type: String, required: true },
    family_id: { type: String, required: true },
    format: { type: String, required: true },
    dots: { type: Number, required: true },
    last_changed: { type: Date, required: true },
    active: { type: Boolean, required: true }
});

const Product = mongoose.model('Product', productSchema);

module.exports = Product;